### PR TITLE
KOGITO-8979 - Data Index PostgreSQL persistence addon

### PR DESCRIPTION
**Pull request by cristianonicolai**
New Quarkus extension to only persist Data Index data without GraphQL endpoint. This is the base change, once approved will follow up with other dbs.

Assembly:
- https://github.com/kiegroup/kogito-apps/pull/1843
- https://github.com/kiegroup/kogito-runtimes/pull/3186

### CODE REVIEW

**Comment by kie-ci3** [comment]
**(tests) - serverless-workflow-examples job** `#687` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>


export BUILD_MVN_OPTS_CURRENT=-Pkogito-apps-downstream 
build-chain build cross_pr -f 'https://raw.githubusercontent.com/kiegroup/kogito-pipelines/main/.ci/pull-request-config.yaml' -o 'bc' -p kiegroup/kogito-examples -u https://github.com/kiegroup/kogito-apps/pull/1843 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-apps.tests.downstream.serverless-workflow-examples/687/display/redirect

**Test results:**
- PASSED: 81
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-apps.tests.downstream.serverless-workflow-examples/687/testReport/org.kie.kogito.examples/DataIndexRestIT/testDataIndexRest/">org.kie.kogito.examples.DataIndexRestIT.testDataIndexRest</a></summary>
Cannot invoke method size() on null object
</details>

**Comment by kie-ci3** [comment]
**(tests) - kogito-apps job** `#1288` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>


export BUILD_MVN_OPTS_CURRENT=-Dvalidate-formatting -Prun-code-coverage
build-chain build cross_pr -f 'https://raw.githubusercontent.com/kiegroup/kogito-pipelines/main/.ci/pull-request-config.yaml' -o 'bc' -p kiegroup/kogito-apps -u https://github.com/kiegroup/kogito-apps/pull/1843 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-apps.tests.kogito-apps/1288/display/redirect

**Test results:**
- PASSED: 2780
- FAILED: 2

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-apps.tests.kogito-apps/1288/testReport/org.kie.kogito.addons.quarkus.data.index.it/PostgreSQLQuarkusAddonDataIndexIT/Build_projects___testDataIndexAddon/">Build projects / org.kie.kogito.addons.quarkus.data.index.it.PostgreSQLQuarkusAddonDataIndexIT.testDataIndexAddon</a></summary>
Cannot invoke method size() on null object
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-apps.tests.kogito-apps/1288/testReport/org.kie.kogito.addons.quarkus.data.index.it/PostgreSQLQuarkusAddonDataIndexIT/Build_projects___testDataIndexAddon/">Build projects / org.kie.kogito.addons.quarkus.data.index.it.PostgreSQLQuarkusAddonDataIndexIT.testDataIndexAddon</a></summary>
Cannot invoke method size() on null object
</details>

**Review by nmirasch — APPROVED** [review]

File: kogito-apps-bom/pom.xml [hunk 1]
```diff
@@ -174,9 +174,25 @@
             </dependency>
             <dependency>
                 <groupId>org.kie.kogito</groupId>
-                <artifactId>data-index-common</artifactId>
```
Comment (radtriste) [inline kogito-apps-bom/pom.xml:177]: why removing `data-index-common` ?

**Review by radtriste — APPROVED** [review]
minor comment

File: kogito-apps-bom/pom.xml (hunk 1 above)
Comment (cristianonicolai) [reply kogito-apps-bom/pom.xml:177]: that's just the test-jar that is not needed anymore since Im adding data-index-test-utils

**Pull request merged**